diff --git a/drivers/input/touchscreen/Kconfig b/drivers/input/touchscreen/Kconfig
index 6a538a082..0521540dc 100644
--- a/drivers/input/touchscreen/Kconfig
+++ b/drivers/input/touchscreen/Kconfig
@@ -1433,3 +1433,11 @@ config TOUCHSCREEN_HIMAX_HX83112B
 	  module will be called himax_hx83112b.
 
 endif
+
+config TOUCHSCREEN_ST_FTS
+    tristate "STMicroelectronics FTS touchscreen"
+    depends on I2C
+    help
+      Say Y to enable support for the STMicroelectronics FTS
+      FingerSense capacitive touchscreen controller (fts_521 series).
+      Used in Xiaomi Mi 9 (cepheus).
diff --git a/drivers/input/touchscreen/Makefile b/drivers/input/touchscreen/Makefile
index 6a7e7c319..147d1725a 100644
--- a/drivers/input/touchscreen/Makefile
+++ b/drivers/input/touchscreen/Makefile
@@ -122,3 +122,4 @@ obj-$(CONFIG_TOUCHSCREEN_IQS7211)	+= iqs7211.o
 obj-$(CONFIG_TOUCHSCREEN_ZINITIX)	+= zinitix.o
 obj-$(CONFIG_TOUCHSCREEN_HIMAX_HX83112B)	+= himax_hx83112b.o
 obj-$(CONFIG_TOUCHSCREEN_NT36523_SPI) += nt36523/
+obj-$(CONFIG_TOUCHSCREEN_ST_FTS)	+= fts_521/
